src/NLog/Targets/FileTarget.cs

```diff
@@ -2183,6 +2196,28 @@ private void WriteFooter(string fileName)
             }
         }
 
+        /// <summary>
+        /// Decision logic whether to archive logfile on startup.
+        /// <see cref="ArchiveOldFileOnStartup"/> and <see cref="ArchiveOldFileOnStartupAboveSize"/> properties.
+        /// </summary>
+        /// <param name="fileName">File name to be written.</param>
+        /// <returns>Decision whether to archive or not.</returns>
+        private bool ShouldArchiveOldFileOnStartup(string fileName)
+        {
+            if (!ArchiveOldFileOnStartup)
```

Comment:
Change this to `if (!ArchiveOldFileOnStartup && ArchiveOldFileOnStartupAboveSize <= 0)`

Reply:
isn't that conflicting with https://github.com/NLog/NLog/pull/3755/files/97b0d69e88637e378fba313af3ac42a3a0664e10#diff-ed0a3fc25ad7408a5b63501fa77dbe69R2212-R2215 ?

Reply:
(PS @snakefoot I could edit the PR also from the Github GUI, I think that also counts for you)

Reply:
I guess the pretty version would be this:

```c#
        private bool ShouldArchiveOldFileOnStartup(string fileName)
        {
            // No size threshold specified
            if (ArchiveOldFileOnStartupAboveSize <= 0)
            {
                 return ArchiveOldFileOnStartup;
            } 
```

Reply:
it's the same behavior, just another version how to write, isn't?

I find it debatable which one is more readable. In terms of performance is no  (significant) difference.

Reply:
> I guess the pretty version would be this:
> 
> ```cs
>         private bool ShouldArchiveOldFileOnStartup(string fileName)
>         {
>             // No size threshold specified
>             if (ArchiveOldFileOnStartupAboveSize <= 0)
>             {
>                  return ArchiveOldFileOnStartup;
>             } 
> ```

Pretty but more complicated in my opinion. My version is straight forward and easy to understand that's why I did it with multiple return statements.

But as I said. Tell me what you want and I'll change it.

Reply:
No need to change IMO